From d4b0847c589702b1eb15925141711cdce6516162 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Thu, 9 Oct 2025 18:31:26 +0800
Subject: [PATCH 024/268] clk: qcom: gcc-sc7280: Add missing GDSC flags

PCIe, UFS and USB GCC GDSCs can and should use the retain registers so
as not to lose their state when entering lower power modes.

Signed-off-by: Xilin Wu <sophon@radxa.com>
---
 drivers/clk/qcom/gcc-sc7280.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/clk/qcom/gcc-sc7280.c
+++ b/drivers/clk/qcom/gcc-sc7280.c
@@ -3102,7 +3102,7 @@ static struct gdsc gcc_pcie_0_gdsc = {
 		.name = "gcc_pcie_0_gdsc",
 	},
 	.pwrsts = PWRSTS_RET_ON,
-	.flags = VOTABLE,
+	.flags = VOTABLE | RETAIN_FF_ENABLE,
 };
 
 static struct gdsc gcc_pcie_1_gdsc = {
@@ -3111,7 +3111,7 @@ static struct gdsc gcc_pcie_1_gdsc = {
 		.name = "gcc_pcie_1_gdsc",
 	},
 	.pwrsts = PWRSTS_RET_ON,
-	.flags = VOTABLE,
+	.flags = VOTABLE | RETAIN_FF_ENABLE,
 };
 
 static struct gdsc gcc_ufs_phy_gdsc = {
@@ -3123,7 +3123,7 @@ static struct gdsc gcc_ufs_phy_gdsc = {
 		.name = "gcc_ufs_phy_gdsc",
 	},
 	.pwrsts = PWRSTS_OFF_ON,
-	.flags = VOTABLE,
+	.flags = VOTABLE | RETAIN_FF_ENABLE,
 };
 
 static struct gdsc gcc_usb30_prim_gdsc = {
@@ -3135,7 +3135,7 @@ static struct gdsc gcc_usb30_prim_gdsc =
 		.name = "gcc_usb30_prim_gdsc",
 	},
 	.pwrsts = PWRSTS_RET_ON,
-	.flags = VOTABLE,
+	.flags = VOTABLE | RETAIN_FF_ENABLE,
 };
 
 static struct gdsc gcc_usb30_sec_gdsc = {
@@ -3144,7 +3144,7 @@ static struct gdsc gcc_usb30_sec_gdsc =
 		.name = "gcc_usb30_sec_gdsc",
 	},
 	.pwrsts = PWRSTS_RET_ON,
-	.flags = VOTABLE,
+	.flags = VOTABLE | RETAIN_FF_ENABLE,
 };
 
 static struct gdsc hlos1_vote_mmnoc_mmu_tbu_hf0_gdsc = {
